pkg/internal/ingress/ingress_test.go

```diff
@@ -0,0 +1,230 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/jetstack/cert-manager/pkg/controller"
+	discoveryfake "github.com/jetstack/cert-manager/test/unit/discovery"
+	"github.com/stretchr/testify/assert"
+	networkingv1 "k8s.io/api/networking/v1"
+	networkingv1beta1 "k8s.io/api/networking/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/informers"
+	kubefake "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/utils/pointer"
+)
+
+func TestFunctionalityAgainstV1(t *testing.T) {
```

Comment:
praise: these three tests (TestFunctionalityAgainstV1, TestFunctionalityAgainstV1Beta1, TestFunctionalityAgainstNone) are very easy to understand, thanks!